examples/conform.cpp

```diff
@@ -0,0 +1,163 @@
+// language governing permissions and limitations under the Apache License.
+//
+
+// Example OTIO script that reads a timeline and then relinks clips
+// to movie files found in a given folder, based on matching names.
+//
+// Demo:
+//
+// % ls -1R
+// editorial_cut.otio
+// media/
+//    shot1.mov
+//    shot17.mov
+//    shot99.mov
+//
+// % conform editorial_cut.otio media conformed.otio
+// Relinked 3 clips to new media.
+// Saved conformed.otio with 100 clips.
+//
+// % diff editorial_cut.otio conformed.otio
+// ...
+
+#include "util.h"
+
+#include <opentimelineio/clip.h>
+#include <opentimelineio/externalReference.h>
+#include <opentimelineio/timeline.h>
+
+#include <iostream>
+
+namespace otio = opentimelineio::OPENTIMELINEIO_VERSION;
+
+// Look for media with this name in this folder.
+std::string find_matching_media(std::string const& name, std::string const& folder)
+{
+    // In this case we're looking in the filesystem.
+    // In your case, you might want to look in your asset management system
+    // and you might want to use studio-specific metadata in the clip instead
+    // of just the clip name.
+    // Something like this:
+    // shot = asset_database->find_shot(
+    //    otio::any_cast<std::map<std::string, std::string> >(clip->metadata()["mystudio"])["shotID"]);
+    // new_media = shot->latest_render("mov");
+    
+    const auto matches = glob(folder, name + ".*");
+    
+    if (matches.size() == 0)
+    {
+        //std::cout << "DEBUG: No match for clip '" << name << "'" << std::endl;
+        return std::string();
+    }
+    if (matches.size() == 1)
+    {
+        return matches[0];
+    }
+    else
+    {
+        std::cout << "WARNING: " << matches.size() << " matches found for clip '" <<
+            name << "', using '" << matches[0] << "'";
+        return matches[0];
+    }
+}
+
+// Look for replacement media for each clip in the given timeline.
+//
+// The clips are relinked in place if media with a matching name is found.
+int conform_timeline(
+    otio::SerializableObject::Retainer<otio::Timeline> const& timeline,
+    std::string const& folder)
+{
+    int count = 0;
+    
+    otio::ErrorStatus error_status;
+    const auto clips = timeline->clip_if(&error_status);
+    if (error_status)
+    {
+        std::cout << error_status.outcome << std::endl;
+        throw error_status;
+    }
+    
+    for (auto clip : clips)
+    {
+        // look for a media file that matches the clip's name
+        const auto new_path = find_matching_media(clip->name(), folder);
+
+        // if no media is found, keep going
+        if (new_path.empty())
+            continue;
+
+        // if we found one, then relink to the new path
+        clip->set_media_reference(new otio::ExternalReference(
+            "file://" + new_path,
+            otio::nullopt // we don't know the available range
+        ));
+        count += 1;
+    }
+    
+    return count;
+}
+
+int main(int argc, char** argv)
```

Comment:
```suggestion
int main(int argc, char** argv) try
```
What's this try you say? See below